### PR TITLE
ci: runner shell sudo doesn't affect redirects SC2024

### PR DESCRIPTION
**Pull request by kwanhur**
### What this PR does / why we need it:

fix ci shell scripts warning `sudo` doesn't affect redirects [SC2024](https://github.com/koalaman/shellcheck/wiki/SC2024)

- use `tee` copies standard input to standard output, instead of `>` to write

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: ci/linux_apisix_current_luarocks_runner.sh [hunk 1]
```diff
@@ -33,8 +33,8 @@ script() {
     sudo rm -rf /usr/local/share/lua/5.1/apisix
 
     # install APISIX with local version
-    sudo luarocks install rockspec/apisix-master-0.rockspec --only-deps > build.log 2>&1 || (cat build.log && exit 1)
-    sudo luarocks make rockspec/apisix-master-0.rockspec > build.log 2>&1 || (cat build.log && exit 1)
+    sudo luarocks install rockspec/apisix-master-0.rockspec --only-deps | sudo tee build.log > /dev/null 2>&1 || (cat build.log && exit 1)
```
Comment (spacewander) [inline ci/linux_apisix_current_luarocks_runner.sh:36]: We don't need sudo to write to `build.log`? So the original code's OK.

Comment (kwanhur) [reply ci/linux_apisix_current_luarocks_runner.sh:36]: The original redirects stdout into `build.log`, it isn't recommended for using `sudo`.

If it needs log into `build.log`, suggest using `tee` to make it.

Comment (spacewander) [reply ci/linux_apisix_current_luarocks_runner.sh:36]: No, use no-sudo user is enough to write log to `build.log`.

File: ci/linux_apisix_master_luarocks_runner.sh [hunk 2]
```diff
@@ -37,7 +37,7 @@ script() {
     cp -r ../utils ./
 
     # install APISIX by luarocks
-    sudo luarocks install $APISIX_MAIN > build.log 2>&1 || (cat build.log && exit 1)
+    sudo luarocks install $APISIX_MAIN | tee build.log > /dev/null 2>&1 || (cat build.log && exit 1)
```
Comment (spacewander) [inline ci/linux_apisix_master_luarocks_runner.sh:40]: There is no need to use `tee` to redirect. The original way is good enough.

**Review by spacewander — CHANGES_REQUESTED** [review]

**Comment by spacewander** [comment]
The SC2024 requires `sudo tee` because `sudo` doesn't affect `>`.
See the example, `echo 3 | sudo tee /proc/sys/vm/drop_caches > /dev/null`.

However, as I mentioned several days ago, we don't need `sudo` to write to `build.log`. There is no need to introduce an extra cmd to do what `>` can do.

**Pull request closed without merging**